### ISSUE TITLE
`mismatched_target_os` false negative when using the `cfg` macro instead of the attribute.

### ISSUE THREAD

**Issue by VZout** [rust-lang/rust-clippy#7839]
Lint name: `mismatched_target_os `

I tried this code:

```rust
fn main() {
    // Clippy throws error
    //#[cfg(linux)]
    //println!("Hello, Linux!");

    // Clippy doesn't care
    if cfg!(linux) {
        println!("Hello, Linux!");
    }
}
```

I expected to see the error `operating system used in target family position` here like you would see if you used the attribute `#[cfg(linux)]`

Instead, clippy doesn't give any error.

### Meta

**Rust version (`rustc -Vv`):**

```
rustc 1.52.1 (9bc8c42bb 2021-05-09)
binary: rustc
commit-hash: 9bc8c42bb2f19e745a63f3445f1ac248fb015e53
commit-date: 2021-05-09
host: x86_64-pc-windows-msvc
release: 1.52.1
LLVM version: 12.0.0
```

**Comment by fanzeyi** [comment]
Agree, I came here to report the same issue. It would be great if clippy can detect this. A very easy mistake to make :)

**Comment by fanzeyi** [comment]
Hi all, I tried to take this on and I have a working draft. I still need to tidy up -- reducing duplicated code and write tests, but I think I got it mostly working :)

<img width="440" alt="image" src="https://user-images.githubusercontent.com/409951/145953051-805d4fe2-e8f6-42de-9133-44eb5096487e.png">
